pennylane/devices/default_tensor.py

```diff
@@ -849,3 +889,85 @@ def execute_and_compute_vjp(
         raise NotImplementedError(
             "The computation of vector-Jacobian product has yet to be implemented for the default.tensor device."
         )
+
+
+@singledispatch
+def apply_operation_core(ops: Operation, device):
+    """Dispatcher for _apply_operation."""
+    device._quimb_circuit.apply_gate(
+        qml.matrix(ops).astype(device._dtype), *ops.wires, parametrize=None
+    )
+
+
+@apply_operation_core.register
+def apply_operation_core_multirz(ops: qml.MultiRZ, device):
+    """Dispatcher for _apply_operation."""
+    apply_operation_core(qml.PauliRot(ops.parameters[0], "Z" * len(ops.wires), ops.wires), device)
+
+
+@apply_operation_core.register
+def apply_operation_core_paulirot(ops: qml.PauliRot, device):
+    """Dispatcher for _apply_operation."""
+    theta = ops.parameters[0]
+    wire_map = dict((w, i) for i, w in enumerate(ops.wires))
+    pw = qml.pauli.string_to_pauli_word(ops._hyperparameters["pauli_word"], wire_map)
+    cospsi = qml.math.cos(0.5 * theta) * copy.deepcopy(device._quimb_circuit.psi)
+    for o in pw:
+        device._quimb_circuit.apply_gate(
+            qml.matrix(o).astype(device._dtype), *o.wires, parametrize=None
+        )
+    device._quimb_circuit._psi = cospsi - 1j * qml.math.sin(0.5 * theta) * device._quimb_circuit.psi
+    device._quimb_circuit._psi.compress(max_bond=device._max_bond_dim, cutoff=device._cutoff)
```

Comment:
This assumes that the state is an MPS (otherwise we simply rely on the `_apply_operation` method at this stage). For my understanding, would it be possible to do something similar for `qml.PauliRot` in case the state is not an MPS?

Reply:
From what I can tell it would not raise an error if you try to add two tensor network vectors. Not sure it is advisable though because we are effectively doubling the computation with this no? (I am assuming quimb would then just contract two separate tensor networks for the computation - might be wrong though)

Reply:
Closing because I've implemented @Qottmann 's idea.